models/concerns/expand_data.rb

```diff
@@ -90,4 +90,10 @@ def provide_original_ids
       element.original_id = element.id
     }
   end
+
+  def provide_original_skills
+    (self.services + self.shipments).each{ |element|
+      element.original_skills = element.skills
```

Comment:
It does work like this. I modified _test_returned_skills_ to double check this.

Reply:
vehicle skills are not returned but like you say, it does not hurt.

Reply:
I was talking about _test_returned_skills_.

In fact since today skills are only modified after clustering then we do Marshal.load and there is no problem. As you suggested I added .dup to avoid any error in the future if we edit skills without using Marshal.load.